openbas-api/src/main/java/io/openbas/injectors/email/service/ImapService.java

```diff
@@ -294,6 +294,7 @@ private void synchronizeBox(Folder inbox, Boolean isSent) throws Exception {
         }
         currentState.setValue(String.valueOf(messageCount));
         settingRepository.save(currentState);
+        inbox.close();
```

Comment:
Maybe it should be better to use the aucloseable interface and change the syncFolders method to

```private void syncFolders() throws Exception {
    try {
        // Sync sent
        try (Folder sentBox = imapStore.getFolder(sentFolder)) {
            sentBox.open(Folder.READ_ONLY);
            synchronizeBox(sentBox, true);
        }

        // Sync received
        for (String listeningFolder : inboxFolders) {
            try (Folder inbox = imapStore.getFolder(listeningFolder)) {
                inbox.open(Folder.READ_ONLY);
                synchronizeBox(inbox, false);
            }
        }
    } catch (MessagingException e) {
        log.warning("Connection failure: " + e.getMessage());
        // Retry logic or rethrow the exception
        retrySyncFolders();
    }
}```

What do you think ?